src/python/pants/backend/go/go_sources/analyze_package/main.go

```diff
@@ -120,6 +130,93 @@ func cleanStringSet(valuesMap map[string]bool) []string {
 	return values
 }
 
+// saveCgo saves the information from the #cgo lines in the import "C" comment.
+// These lines set CFLAGS, CPPFLAGS, CXXFLAGS and LDFLAGS and pkg-config directives
+// that affect the way cgo's C code is built.
+func saveCgo(filename string, pkg *Package, cg *ast.CommentGroup, buildContext *build.Context) error {
+	text := cg.Text()
+	for _, line := range strings.Split(text, "\n") {
+		orig := line
+
+		// Line is
+		//      #cgo [GOOS/GOARCH...] LDFLAGS: stuff
+		//
+		line = strings.TrimSpace(line)
+		if len(line) < 5 || line[:4] != "#cgo" || (line[4] != ' ' && line[4] != '\t') {
+			continue
+		}
+
+		// Split at colon.
+		line, argstr, ok := stringsCut(strings.TrimSpace(line[4:]), ":")
+		if !ok {
+			return fmt.Errorf("%s: invalid #cgo line: %s", filename, orig)
+		}
+
+		// Parse GOOS/GOARCH stuff.
+		f := strings.Fields(line)
+		if len(f) < 1 {
+			return fmt.Errorf("%s: invalid #cgo line: %s", filename, orig)
+		}
+
+		cond, verb := f[:len(f)-1], f[len(f)-1]
+		if len(cond) > 0 {
+			ok := false
+			for _, c := range cond {
+				if matchAuto(buildContext, c, nil) {
+					ok = true
+					break
+				}
+			}
+			if !ok {
+				continue
+			}
+		}
+		args, err := splitQuoted(argstr)
+		if err != nil {
+			return fmt.Errorf("%s: invalid #cgo line: %s", filename, orig)
+		}
+		// PANTS NOTE: In the original Go code, this code would have expanded `${SRCDIR}` in a path
+		// to the absolute directory where the cgo file was located. Given Pants uses an execution sandbox,
+		// this will be done when actually building.
+		//for i, arg := range args {
+		//	if arg, ok = expandSrcDir(arg, di.Dir); !ok {
+		//		return fmt.Errorf("%s: malformed #cgo argument: %s", filename, arg)
+		//	}
+		//	args[i] = arg
+		//}
+
+		// PANTS NOTE: In the original Go code, this code would have expanded paths passed to -I and -L compiler
+		// options to be absolute. Given Pants uses an execution sandbox, this will be done when actually building.
+		//
+		// PANTS TODO: Warn the user if the file referenced by a relative path was not included via
+		// `files` or `resources`?
+		//
+		//switch verb {
+		//case "CFLAGS", "CPPFLAGS", "CXXFLAGS", "FFLAGS", "LDFLAGS":
+		//	// Change relative paths to absolute.
+		//	ctxt.makePathsAbsolute(args, di.Dir)
+		//}
```

Comment:
This is commented out original code? If so, it'd help to move those closer to the `PANTS_NOTE` and maybe say "we comment this out and replace it with the below 10 lines". Or maybe better is to delete it.